pkg/table-router/router.go

```diff
@@ -23,10 +24,17 @@ import (
 // TableRule is a rule to route schema/table to target schema/table
 // pattern format refers 'pkg/table-rule-selector'
 type TableRule struct {
-	SchemaPattern string `json:"schema-pattern" toml:"schema-pattern" yaml:"schema-pattern"`
-	TablePattern  string `json:"table-pattern" toml:"table-pattern" yaml:"table-pattern"`
-	TargetSchema  string `json:"target-schema" toml:"target-schema" yaml:"target-schema"`
-	TargetTable   string `json:"target-table" toml:"target-table" yaml:"target-table"`
+	SchemaPattern       string               `json:"schema-pattern" toml:"schema-pattern" yaml:"schema-pattern"`
+	TablePattern        string               `json:"table-pattern" toml:"table-pattern" yaml:"table-pattern"`
+	TargetSchema        string               `json:"target-schema" toml:"target-schema" yaml:"target-schema"`
+	TargetTable         string               `json:"target-table" toml:"target-table" yaml:"target-table"`
+	TableTransferColumn *TableTransferColumn `json:"table-transfer-column" toml:"table-transfer-column" yaml:"table-transfer-column"`
+}
+
+// TableTransferColumn is a rule that converts table names to column values
+type TableTransferColumn struct {
+	TargetColumn string `json:"target-column" toml:"target-column" yaml:"target-column"`
+	TableRegular string `json:"table-regular" toml:"table-regular" yaml:"table-regular"`
```

Comment:
this name is confusing, how about rename it to `extract-table`, `table-extractor`

> There are only two hard things in Computer Science: cache invalidation and naming things.

BTW, would you like to add a `schema-extractor` even if you don't use it? (and we'll find a good place to add `source-extractor` in future)

Reply:
```
type TableRule struct {
	SchemaPattern   string           `json:"schema-pattern" toml:"schema-pattern" yaml:"schema-pattern"`
	TablePattern    string           `json:"table-pattern" toml:"table-pattern" yaml:"table-pattern"`
	TargetSchema    string           `json:"target-schema" toml:"target-schema" yaml:"target-schema"`
	TargetTable     string           `json:"target-table" toml:"target-table" yaml:"target-table"`
	TableExtractor  *TableExtractor  `json:"extract-table" toml:"extract-table" yaml:"extract-table"`
	SchemaExtractor *SchemaExtractor `json:"extract-schema" toml:"extract-schema" yaml:"extract-schema"`
	SourceExtractor *SourceExtractor `json:"extract-source" toml:"extract-source" yaml:"extract-source"`
}

// TableExtractor is a rule that converts table names to column values
type TableExtractor struct {
	TargetColumn string `json:"target-column" toml:"target-column" yaml:"target-column"`
	TableRegular string `json:"table-regular" toml:"table-regular" yaml:"table-regular"`
}
type SchemaExtractor struct {
	TargetColumn  string `json:"target-column" toml:"target-column" yaml:"target-column"`
	SchemaRegular string `json:"schema-regular" toml:"table-regular" yaml:"table-regular"`
}
type SourceExtractor struct {
	TargetColumn  string `json:"target-column" toml:"target-column" yaml:"target-column"`
	SchemaRegular string `json:"schema-regular" toml:"table-regular" yaml:"table-regular"`
}
```

How does it look now? Maybe schema and source also need to be extracted into columns, is that your means?

Reply:
how about calling `TableTransferColumn `  `TargetVirtualShardingKey`

Reply:
@yufan022 

> Maybe schema and source also need to be extracted into columns, is that your means?

That's a good design, my original idea is to handle this case:

user may need to attach extra column by its source id, or dabatase name, or table name. For example, when merging `us-west.user` table and `us-east.user` table, the database name may need to be attached. 

My original idea is there's only one extra column, and the format maybe configurated such as `{source-extractor}-{schema-extractor}-{table-extractor}`. Your idea is add (up to) three extra columns, I think that's better because we don't need to write such a complicated formatter string. 👍

>  How does it look now?

the `regular` in `table-regular` seems unfamiliar to me, maybe other users won't recognize it as regular expression, can you give this usage in other software configurations? BTW I prefer `table-regexp`

And above all, if you just want to make a quick contribution and don't want to pay more time at over-design for now, we can move forward!

@Ehco1996 

> how about calling TableTransferColumn TargetVirtualShardingKey

that's too long, I don't like fields more than two words 😂

Reply:
>that's too long, I don't like fields more than two word

ok, we can find anothor name for this field

> Your idea is add (up to) three extra columns  I think that's better because we don't need to write such a complicated formatter string. 

I'm a bit worried about this design, as some of the work afterwards will rely on these (three) fields for UK
If I use a joint unique index of three fields, I'm worried that there will be performance issues when writing @lance6716

Reply:
> the regular in table-regular seems unfamiliar to me, maybe other users won't recognize it as regular expression, can you give this usage in other software configurations? BTW I prefer table-regexp

`table-regexp` looks better.

> And above all, if you just want to make a quick contribution and don't want to pay more time at over-design for now, we can move forward!

I'm glad to see this feature more standardized, So you can put forward more ideas, and I will try to improve it.

Reply:
I actually prefer to generate one column rather than three. The newly added column acts as an identifier rather than part of the business data, and more columns may be too intrusive for the user to bother with

Regarding the naming, I am nicknamed "The Black Hole of Naming", my opinion is for reference.

First of all, it must be very clearly distinguished from the existing parameters, i.e. `target-table`, `target-schema`. Because of the secondary frequency of use, must not be similar to the common parameters lead to misuse.

Secondly, it is not sure whether other columns will be added for identification in the future, so for scalability reasons, perhaps we can use the `label_` + `usage` format. Meaning that all columns in this category belong to `label`, this time for labeling the source, and next time add columns for labeling new uses